src/components/App/App.js

```diff
@@ -4,38 +4,54 @@ import treeData from '../../assets/treeData';
 import TreeDisplay from '../TreeDisplay/TreeDisplay';
 import Header from '../Header/Header';
 import Form from '../Form/Form';
+import { Route } from 'react-router-dom';
 
 class App extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      trees: []
+      trees: [],
+      currentTree: {}
     }
   }
 
   componentDidMount() {
-    this.setState({ trees: treeData })
+    this.setState({ 
+      trees: treeData,
+      currentTree: treeData[this.getRandomIndex(0, treeData.length - 1)] 
+    })
   }
 
-  getRandomTree = (min, max) => {
+  getRandomIndex = (min, max) => {
       return Math.floor(Math.random() * (max - min + 1) + min);
   }
 
   addTree = (newTree) => {
     this.setState({ trees: [...this.state.trees, newTree] })
   }
 
+  setNewTree = () => {
+    this.setState({ currentTree: this.state.trees[this.getRandomIndex(0, treeData.length - 1)] })
+  }
+
   render() {
+    if (this.state.trees.length) {
     return (
       <>
-        <Header />
-        {
-        this.state.trees.length &&
-        <TreeDisplay tree={this.state.trees[this.getRandomTree(0, this.state.trees.length)]} />
-        }
-        <Form addTree={this.addTree} />
+        <Header setNewTree={this.setNewTree}/>
+        <Route exact path='/'>
+          {this.state.trees.length &&
+          <TreeDisplay tree={this.state.currentTree} />
+          }
+        </Route>
+        <Route path='/addtree'> 
+          <Form addTree={this.addTree} />
+        </Route>
       </>
```

Comment:
At a later refactor stage it might be good to add a Switch and then create a 404 page for nonexistent routes. Just to add some extra flair!